### ISSUE TITLE
Docs: Automatically inject rule option descriptions into their docs

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9867]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Following #5392, most rule options will have `description`s in their schemas. For many of those options, their docs either:

* Don't have an actual description of the option (see the `allowAny` example later)
* As of #9868, roughly -or exactly- match what's on their docs pages

Proposal: let's have the rule docs generation also inject the `description` and default value for each rule option into its `###` heading?

For example, the `allowAny` `description` in `restrict-plus-operands`:

https://github.com/typescript-eslint/typescript-eslint/blob/ce79c552da146eeb565a1be37f2568ba4f281881/packages/eslint-plugin/src/rules/restrict-plus-operands.ts#L61-L62

...could be used as a first line in the docs page under its h3:

https://github.com/typescript-eslint/typescript-eslint/blob/a7bdd1c1eb3a403c919e516d58b1d667ea23219a/packages/eslint-plugin/docs/rules/restrict-plus-operands.mdx#L75-L77

We could also mention the default value.

In other words, the end result I'm proposing is:

```diff
### `allowAny`

+ Whether to allow `any` typed values.
+ Default: `false`.

Examples of code for this rule with `{ allowAny: true }`:
```

We'd have to do a pass to remove redundant lines in the `.mdx` files.

### Affected URL(s)

https://typescript-eslint.io/rules/*

💖
